### ISSUE TITLE
Make Vagrant binary a parameter

### ISSUE THREAD

**Issue by bheuvel** [test-kitchen/kitchen-vagrant#218]
If I'm going wrong about this, please let me know (how otherwise).

tl;dr the proposed PR #219 solves my problem, basically enables running Vagrant from master (in/from kitchen off course :wink:) on Windows.

**Problem:**
Due to dependencies I was forced to upgrade to Vagrant >= 1.8.
Vagrant 1.8 (msi/installer) introduces Ruby 2.2, which does not support the (msi) bundled Nokogiri, thereby making Vagrant >= 1.8 inoperable for me
This has been fixed on master (details: https://github.com/mitchellh/vagrant/pull/6848)

**Solution attempt 1, bundle**
Including Vagrant (master) in my bundler config (gem via github)
**Fail**
This does not work as kitchen calls `vagrant` using `Mixlib::ShellOut`: https://github.com/test-kitchen/test-kitchen/blob/master/lib/kitchen/shell_out.rb#L67
Bundler does not pick this up, resulting in a Vagrant not found error.

**Solution attempt 2, binstubs**
To use very latest Vagrant master, the wiki explains how to do this: https://github.com/mitchellh/vagrant/wiki/Installing-Vagrant-from-Source#use-dev-version.
Basically Bundler creates a binstub `vagrant`, which is a ruby file (shebang).
**Fail**
These shebang files do not work on Windows (cmd/PowerShell). (_Just calling the file doesn't work. Calling `ruby <vagrant_stub>` does work_)

**Solution attempt 2.1, Cygwin using the binstub**
When using Cygwin or mingw, the shebang works, and the latest development version of Vagrant is available when called directly from a bash shell.
**Fail**
But kitchen-vagrant uses [`Mixlib::ShellOut`](https://github.com/test-kitchen/test-kitchen/blob/master/lib/kitchen/shell_out.rb#L67), which does not use the shell it originated from but uses Win32API which is _somewhat_ equivalent to the default shell, `cmd`, where the shebang binstubs don't work.

**Solution attempt 2.2, cmd file using the binstub**
Considering the fact that the binstubs do work in _native_ Windows when using `ruby <full path to the binstub>`, I created a `vagrant.cmd` file with content: `ruby c:\dev\vagrant\exec\vagrant %*`, and put that in my path.
Thereby from cmd or PowerShell able to use e.g `vagrant --version`.
**Fail**
But kitchen-vagrant uses `Mixlib::ShellOut`, which requires that exact command (`vagrant`), or _command_.EXE (`vagrant.exe`) to exist, and does not pick up `vagrant.cmd`

**Solution attempt 2.2.1, code patch using cmd file using the binstub**
To be able to make attempt 2.2 work, I would need to change the kitchen-vagrant code, like follows:
from
```
self.class.vagrant_version ||= run_silently(
  "vagrant --version", :cwd => Dir.pwd).chomp.split(" ").last
```
to
```
self.class.vagrant_version ||= run_silently(
  "vagrant.cmd  --version", :cwd => Dir.pwd).chomp.split(" ").last
```

**Success!**

With PR #219, I have made `vagrant` a parameter, which could be for example `vagrant.cmd`, but can also be something like `ruby c:\dev\vagrant\exec\vagrant` (both work).

**Comment by bheuvel** [comment]
Implemented with #219
